### PR TITLE
Throw an error instead of warning on Softmax Implicit dimension

### PR DESCRIPTION
**Pull request by AmeenAli**
Hello,
I have been debugging my model for hours until I realized I'm doing softmax along the wrong dimension because I didn't specify the Dim implicitly in Softmax argument and Apparently i have missed the warning message because i was printing lot of stuff on the way.
So i'm suggesting to make it raise an error instead of warning, i'm sure this will save people time

### CODE REVIEW

**Comment by ngimel** [comment]
This is a backward-compatibility breaking behavior, so we cannot merge this PR. However, the warning behavior and how to make warnings not being lost in the output spew is being discussed here #37315

**Comment by mruberry** [comment]
Sorry the warning was tough to spot, @AmeenAli, and thank you for submitting a PR attempting to address the issue. As @ngimel says, however, this has the potential to break older PyTorch programs that rely on the current behavior, so we can't accept this PR at this time. We'd need to do more research into its impact.
